git_machete/client.py

```diff
@@ -731,6 +731,8 @@ def traverse(
                 needs_remote_sync = True
                 if branch in self.annotations:
                     needs_remote_sync = self.annotations[branch].qualifiers.push
+                if not opt_push_tracked and not opt_push_untracked:
```

Comment:
Pls add a test that would capture the difference b/w the old and the new behavior